crates/ra_ide/src/runnables.rs

```diff
@@ -16,6 +16,7 @@ use std::fmt::Display;
 pub struct Runnable {
     pub range: TextRange,
     pub kind: RunnableKind,
+    pub features_needed: Option<Vec<SmolStr>>,
```

Comment:
`features` is a cargo-specific concept. `ra_ide` layer doesn't know about `Cargo`, so it shouldn't talk about features directly. I think here we should store a `CfgExpr`, and then extract features out of it at the `rust-analyzer` layer.